airflow/contrib/operators/s3_delete_objects_operator.py

```diff
@@ -0,0 +1,88 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from pprint import pformat
+
+from airflow.hooks.S3_hook import S3Hook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+
+
+class S3DeleteObjectsOperator(BaseOperator):
+    """
+    To enable users to delete single object or multiple objects from
+    a bucket using a single HTTP request.
+
+    Users may specify up to 1000 keys to delete.
+
+    :param bucket: Name of the bucket in which you are going to delete object(s)
+    :type bucket: str
+    :param keys: The key(s) to delete from S3 bucket.
+
+        When ``keys`` is a string, it's supposed to be the key name of
+        the single object to delete.
+
+        When ``keys`` is a list, it's supposed to be the list of the
+        keys to delete.
+    :type keys: str or list
+    :param s3_conn_id: Connection id of the S3 connection to use
+    :type s3_conn_id: str
+    :param verify: Whether or not to verify SSL certificates for S3 connetion.
+        By default SSL certificates are verified.
+
+        You can provide the following values:
+
+        - False: do not validate SSL certificates. SSL will still be used,
+                 but SSL certificates will not be
+                 verified.
+        - path/to/cert/bundle.pem: A filename of the CA cert bundle to uses.
+                 You can specify this argument if you want to use a different
+                 CA cert bundle than the one used by botocore.
+    :type verify: bool or str
+    """
+
+    @apply_defaults
+    def __init__(
+            self,
+            bucket,
+            keys,
+            s3_conn_id='aws_default',
+            verify=None,
+            *args, **kwargs):
+        super(S3DeleteObjectsOperator, self).__init__(*args, **kwargs)
+        self.bucket = bucket
+        if isinstance(keys, list):
+            self.keys = keys
+        else:
+            self.keys = [keys]
+        self.s3_conn_id = s3_conn_id
+        self.verify = verify
+
+    def execute(self, context):
+        s3_hook = S3Hook(aws_conn_id=self.s3_conn_id, verify=self.verify)
+
+        delete_dict = {"Objects": [{"Key": k} for k in self.keys]}
+        response = s3_hook.get_conn().delete_objects(Bucket=self.bucket,
+                                                     Delete=delete_dict)
+
+        deleted_keys = [x['Key'] for x in response.get("Deleted", [])]
+        errors_keys = [x['Key'] for x in response.get("Errors", [])]
+        self.log.info("Deleted: {}".format(deleted_keys))
+        self.log.info("Errors: {}".format(errors_keys))
```

Comment:
If there are errors then then task should fail.

Reply:
`Errors` here appear only because users specified some keys to delete while these keys don't exist in the bucket. It doesn't have to mean an actual error.

If we fail on "error" here, implicitly we're adding a requirement: you can only delete objects that you're 100% sure that they exist in the bucket.

Or possibly we can add another argument, say `fail_if_not_found`. Only when it's set to `True`, the task would fail when the key to delete doesn't exist. (default value should be?)

Any thoughts on this?

Reply:
Ah right, if that's the only time it can fail then this is the right behaviour - failing to delete a file that doesn't exist is not an error, but what about other cases, such as failing to delete because the object is owned by another user (something I run across frequently with cross-account bucket set ups))

Reply:
I believe the case you described would return an exception instead, while the `Errors` I mention here is an entry in the response from a successful request.

On the other hand, I'll double-confirm whether `failing to delete a file that doesn't exist` is the only case causing `Errors` entry in the response from a successful request.